packages/terra-application/src/primary-navigation-layout/workspace-layout/WorkspaceLayout.jsx

```diff
@@ -0,0 +1,461 @@
+
+            userSelectedTypeRef.current = undefined;
+
+            if (scale >= 1) {
+              userSelectedScaleRef.current = 1.0;
+
+              setWorkspaceSize({
+                scale: 1.0,
+                type: undefined,
+              });
+            } else if (scale < 0) {
+              userSelectedScaleRef.current = 0;
+
+              setWorkspaceSize({
+                scale: 0,
+                type: undefined,
+              });
+            } else {
+              userSelectedScaleRef.current = scale;
+
+              setWorkspaceSize({
+                scale,
+                type: undefined,
+              });
+            }
+          }}
+        />
+      </div>
+    );
+  };
+
+  const renderWorkspaceOverlay = () => {
+    if (!workspaceIsVisible || !hasOverlayWorkspace) {
+      return undefined;
+    }
+
+    return (
+      <div
+        role="none"
+        ref={workspaceOverlayRef}
+        className={cx('workspace-overlay')}
+        onClick={() => { setWorkspaceIsVisible(false); }} // eslint-disable-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+      />
+    );
+  };
+
+  const renderContent = () => (
+    <div
+      id={`${id}-content-body`}
+      className={cx('content-body')}
+      ref={contentElementRef}
+      style={workspaceSize.scale !== undefined && workspaceIsVisible ? { flexGrow: `${1 - workspaceSize.scale}`, msFlexPositive: `${1 - workspaceSize.scale}` } : null} // eslint-disable-line react/forbid-dom-props
+      inert={hasOverlayWorkspace && workspaceIsVisible ? 'true' : null}
+    >
+      <LayoutActionsContext.Provider value={layoutActionsContextValue}>
+        {children}
+      </LayoutActionsContext.Provider>
+    </div>
+  );
+
+  const renderWorkspace = () => {
+    if (!workspace) {
+      return undefined;
+    }
+
+    return (
+      <>
+        {renderResizeHandle()}
+        <div
+          id={`${id}-workspace-body`}
+          ref={workspacePanelRef}
+          className={cx('workspace-body', { visible: workspaceIsVisible, overlay: hasOverlayWorkspace })}
+          style={workspaceSize.scale !== undefined ? { flexGrow: `${workspaceSize.scale}` } : null} // eslint-disable-line react/forbid-dom-props
+          tabIndex="-1"
+          aria-labelledby={`${id}-workspace-container`}
+        >
+          <div
+            className={cx('workspace-inner')}
+          >
+            {clonePropsOntoWorkspace()}
+          </div>
+        </div>
+        {renderWorkspaceOverlay()}
+      </>
+    );
+  };
+
+  return (
+    <>
+      {renderWorkspaceSkipToLink()}
+      <div
+        id={id}
+        className={cx('layout-container', { 'workspace-visible': workspaceIsVisible, [`workspace-${workspaceSize.size}`]: workspaceSize.size && !workspaceSize.px, [`workspace-${workspaceSize.type}`]: workspaceSize.type && !workspaceSize.px })}
+        ref={pageContainerRef}
```

Comment:
Could rename this ref to be more clearly WorkspaceLayout-related.

Reply:
Updated.